app/main.py

```diff
@@ -41,12 +63,48 @@ def read_user(e_mail: str, db: Session = Depends(get_db)):
 
 
 
+@app.post("/tasks/", response_model=TaskOut)
+def task_maker(task: TaskCreate, db: Session = Depends(get_db), user: UserOut = Depends(get_current_user)):
+    return create_task(db, task, user.email_address)
 
+@app.get("/tasks/", response_model=List[TaskOut])
+def read_tasks(db: Session = Depends(get_db), user: UserOut = Depends(get_current_user)):
+    return get_tasks(db=db, user_id=user.email_address)
 
+@app.get("/tasks/priority_first/", response_model=List[TaskOut]) 
+def filter_tasks_by_priority(db: Session = Depends(get_db), user: UserOut = Depends(get_current_user)):
+    return get_tasks_by_priority(db=db, user_id=user.email_address)
 
+@app.get("/tasks/status_choose/{status}", response_model=List[TaskOut])
+def filter_tasks_by_status(status: TaskStatus, db: Session = Depends(get_db), user: UserOut = Depends(get_current_user)):
+    return get_tasks_by_status(db=db, user_id=user.email_address, status=status)
 
+@app.get("/tasks/deadline_first", response_model=List[TaskOut])
+def filter_tasks_by_deadline(db: Session = Depends(get_db), user: UserOut = Depends(get_current_user)):
+    return get_task_by_deadline(db=db, user_id=user.email_address)
 
-
+@app.get("/tasks/{task_id}", response_model=TaskOut)
+def read_task(task_id: int, db: Session = Depends(get_db), user: UserOut = Depends(get_current_user)):
+    try:
+        task = get_task(db=db, task_id=task_id, user_id=user.email_address)
+        return task
+    except Exception as e:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail=str(e))
```

Comment:
So what ever happens inside task, even like a task is not found with that id, it will return 401? That can't be right am I missing smtn